assets/src/editor/events/hocs/with-price-type-entities.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * External imports
+ */
+import { withSelect } from '@wordpress/data';
+import { createHigherOrderComponent } from '@wordpress/compose';
+
+/**
+ * withPriceTypeEntities
+ * returns an object containing the following:
+ *    priceTypeEntities - an array of price_type entities
+ *    priceTypeEntitiesLoaded - boolean true if loading is complete
+ *
+ * @function
+ */
+export default createHigherOrderComponent(
+	withSelect(
+		( select ) => {
+			const { getEntities } = select( 'eventespresso/lists' );
+			const priceTypeEntities = getEntities( 'price_type' );
+			const { hasFinishedResolution } = select( 'core/data' );
+			const priceTypeEntitiesLoaded = hasFinishedResolution(
+				'eventespresso/lists',
+				'getEntities',
+				[ 'price_type' ]
+			);
+			return { priceTypeEntities, priceTypeEntitiesLoaded };
+		}
+	),
```

Comment:
This still loads price type entities from the server on demand.  I think this is something that we _could_ move to the initialization layer but I don't think it creates too much of a problem having it load on demand either.